tutorials-and-examples/storage/hyperdisk-ml/README.md

```diff
@@ -0,0 +1,207 @@
+```
+
+7. In your GKE cluster create your Hypedisk ML multi zone and standard Storage Classes. Hyperdisk ML disks are zonal and the Hyperdisk-ml-multi-zone storage class automatically provisions disks in zones where the pods using them are. 
+Replace the zones in this class with the zones you want to allow the Hyperdisk ML snapshot to create disks in. 
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: hyperdisk-ml-multi-zone
+parameters:
+  type: hyperdisk-ml
+  provisioned-throughput-on-create: "2400Mi"
+  enable-multi-zone-provisioning: "true"
+provisioner: pd.csi.storage.gke.io
+allowVolumeExpansion: false
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+allowedTopologies:
+- matchLabelExpressions:
+  - key: topology.gke.io/zone
+    values:
+    - us-central1-a
+    - us-central1-c
+--- 
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+    name: hyperdisk-ml
+parameters:
+    type: hyperdisk-ml
+provisioner: pd.csi.storage.gke.io
+allowVolumeExpansion: false
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+```
+
+7. Create a volumeSnapshotClass and VolumeSnapshotContent config to use your snapshot. Replace the VolumeSnapshotContent.spec.source.snapshotHandle with the path to your snapshot. 
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: my-snapshotclass
+driver: pd.csi.storage.gke.io
+deletionPolicy: Delete
+---
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: restored-snapshot
+spec:
+  volumeSnapshotClassName: my-snapshotclass
+  source:
+    volumeSnapshotContentName: restored-snapshot-content
+---
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotContent
+metadata:
+  name: restored-snapshot-content
+spec:
+  deletionPolicy: Retain
+  driver: pd.csi.storage.gke.io
+  source:
+    snapshotHandle: projects/[project_ID]/global/snapshots/[snapshotname]
+  volumeSnapshotRef:
+    kind: VolumeSnapshot
+    name: restored-snapshot
+    namespace: default
+
+```
+
+8. Reference your snapshot in the persistent volume claim. Be sure to adjust the spec.dataSource.name to your snapshot name
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: hdml-consumer-pvc
+spec:
+  dataSource:
+    name: restored-snapshot
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+  - ReadOnlyMany
+  storageClassName: hyperdisk-ml-multi-zone
+  resources:
+    requests:
+      storage: 140Gi
+```
+
+8. Add a reference to this PVC in your deployment spec.template.spec.volume.persistentVolumeClaim.claimName parameter. 
+
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ubuntu
+  labels:
+    app: ubuntu
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ubuntu
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: ubuntu
+    spec:
+      containers:
+      - image: bkauf/storage:latest
```

Comment:
I'd just use `busybox` as the image here here, as it's a small container image, and has `sleep`